lib/chef/knife/ec2_base.rb

```diff
@@ -118,18 +118,22 @@ def is_image_windows?
       def validate!(keys=[:aws_access_key_id, :aws_secret_access_key])
         errors = []
 
-        unless Chef::Config[:knife][:aws_config_file].nil?
-          aws_config = ini_parse(File.read(Chef::Config[:knife][:aws_config_file]))
-          profile = if !Chef::Config[:knife][:aws_profile].nil? then 'profile '+Chef::Config[:knife][:aws_profile] else  'default' end
-          
-          unless aws_config.values.empty? 
+        if locate_config_value(:aws_config_file)
+          aws_config = ini_parse(File.read(locate_config_value(:aws_config_file)))
+          profile = (locate_config_value(:aws_profile) == 'default') ? 'default' : 'profile '+locate_config_value(:aws_profile)
```

Comment:
please drop the ternary operator here and expand this to multiple lines for readability, e.g.

```
profile = if locate_config_value(:aws_profile) == 'default'
                 'default'
               else
                 "profile #{locate_config_value(:aws_profile}"
               end
```